src/filtertransdialog.cpp

```diff
@@ -1423,20 +1423,10 @@ template <class MODEL, class DATA> bool mmFilterTransactionsDialog::mmIsRecordMa
 {
     bool ok = true;
 
-    wxString refType;
-    // Check the Data type to determine the tag RefType
-    if (typeid(tran).hash_code() == typeid(Model_Checking::Data).hash_code())
-        refType = Model_Attachment::reftype_desc(Model_Attachment::TRANSACTION);
-    else if (typeid(tran).hash_code() == typeid(Model_Billsdeposits::Data).hash_code())
-        refType = Model_Attachment::reftype_desc(Model_Attachment::BILLSDEPOSIT);
-
     // wxLogDebug("Check date? %i trx date:%s %s %s", getDateRangeCheckBox(), tran.TRANSDATE, getFromDateCtrl().GetDateOnly().FormatISODate(),
-    wxDate date;
-    date.ParseDateTime(tran.TRANSDATE) || date.ParseDate(tran.TRANSDATE);
-    wxString strDate = date.FormatISOCombined();
     if (mmIsAccountChecked() && m_selected_accounts_id.Index(tran.ACCOUNTID) == wxNOT_FOUND && m_selected_accounts_id.Index(tran.TOACCOUNTID) == wxNOT_FOUND)
         ok = false;
-    else if ((mmIsDateRangeChecked() || mmIsRangeChecked()) && (strDate < m_begin_date || strDate > m_end_date))
+    else if ((mmIsDateRangeChecked() || mmIsRangeChecked()) && (tran.TRANSDATE < m_begin_date || tran.TRANSDATE > m_end_date))
```

Comment:
We can't compare straight TRANSDATE here because 'YYYY-mm-dd' < 'YYYY-mm-ddT00:00:00', so all transactions without a time are always filtered out even if they fit the date range. We need to format TRANSDATE as ISOCombined prior to checking against the specified filter values.

Reply:
> 'YYYY-mm-dd' < 'YYYY-mm-ddT00:00:00'

![image](https://github.com/moneymanagerex/moneymanagerex/assets/6836805/f7f2aaed-e72a-4283-aef1-b8064a32bb73)

Yes, it's just that dates without specifying the time will always be at the top of the list.

Reply:
https://github.com/moneymanagerex/moneymanagerex/pull/6555/commits/7482dd1e22a2d0a4cacf4108855dcf036a754489

Here is the solution: if the time is not set, then we assume that it is any time of the day and not 00:00:00

Reply:
That works, though I feel it limits the usefulness of the filter -- there is no way to isolate transactions with a specific time, the report shows everything from that day with no time as well. You can't filter to show transactions on 2024-03-05 with a time between 13:00 and 16:00.

Personally I prefer treating no time the same as 00:00:00. If the user is modifying the time field on the date filter then they are not looking for all transactions on that date, otherwise they would leave the default range 00:00:00 - 23:59:59.

Maybe others want to weigh in here.

Reply:
> Personally I prefer treating no time the same as 00:00:00. If the user is modifying the time field on the date filter then they are not looking for all transactions on that date, otherwise they would leave the default range 00:00:00 - 23:59:59.

I would concur. In fact I suspect that most of the users (me included) don't even need to record the transaction time.